src/leetCodeExplorer.ts

```diff
@@ -64,57 +73,100 @@ export class LeetCodeTreeDataProvider implements vscode.TreeDataProvider<LeetCod
+                        name: "Difficulty",
+                    }), false),
+                    new LeetCodeNode(Object.assign({}, list.IProblemDefault, {
+                        id: "Root",
+                        name: "Tag",
+                    }), false),
+                    new LeetCodeNode(Object.assign({}, list.IProblemDefault, {
+                        id: "Root",
+                        name: "Company",
+                    }), false),
+                    new LeetCodeNode(Object.assign({}, list.IProblemDefault, {
+                        id: "Root",
+                        name: "Favorite",
+                    }), false),
+                ]);
             });
         } else {
-            return element.isProblem ? [] : this.composeProblemNodes(element.name);
+            switch (element.name) { // First-level
+                case "Difficulty":
+                case "Tag":
+                case "Company":
+                    return this.composeCategoryNodes(element);
+                case "Favorite":
+                    return this.treeData.Favorite.map((p: list.IProblem) => new LeetCodeNode(p));
+                default: // Second and lower levels
+                    return element.isProblem ? [] : this.composeProblemNodes(element);
+            }
         }
     }
 
     private async getProblemData(): Promise<void> {
-        const allProblems: list.IProblem[] = await list.listProblems();
-        this.treeData.clear();
-        for (const problem of allProblems) {
-            const problems: list.IProblem[] | undefined = this.treeData.get(problem.difficulty);
-            if (problems) {
-                problems.push(problem);
-            } else {
-                this.treeData.set(problem.difficulty, [problem]);
+        this.treeData = {
+            Difficulty: new Map(),
+            Tag: new Map(),
+            Company: new Map(),
+            Favorite: [],
+        };
+        for (const problem of await list.listProblems()) {
+            // Add favorite problem, no matter whether it is solved.
+            if (problem.favorite) {
+                this.treeData.Favorite.push(problem);
+            }
+            // Hide solved problem in category view is specified in option.
+            if (problem.state === ProblemState.AC && this.leetCodeConfig.get<boolean>("hideSolved")) {
+                continue;
+            }
+            // Add problems according to category
+            const categories: Array<[Category, string[]]> = [
+                ["Difficulty", [problem.difficulty]],
+                ["Tag", problem.tags],
+                ["Company", problem.companies],
+            ];
+            for (const [parent, children] of categories) {
+                for (let subCategory of children) {
+                    // map 'first-second' to 'First Second'
+                    subCategory = subCategory.split("-").map((c: string) => c[0].toUpperCase() + c.slice(1)).join(" ");
```

Comment:
`first-second` is not that bad.

Cuz current logic could handle `a-b` but not `a-b-c`

Just leave `first-second` as it is for now.

Reply:
What does `a-b-c` refer to? The code here could transform `breadth-first-search` to `Breadth First Search`.

Reply:
My bad, it's fine.